packages/core/src/lib/waku_store/index.ts

```diff
@@ -270,7 +277,8 @@ async function* paginate<T extends DecodedMessage>(
   connection: Connection,
   protocol: string,
   queryOpts: Params,
-  decoders: Map<string, Decoder<T>>
+  decoders: Map<string, Decoder<T>>,
+  cursor?: proto.Index
```

Comment:
```suggestion
  cursor?: Index
```

Would be cleaner if it was the `Index` from `@waku/interfaces`. We should avoid exposing the prototypes to the user.